boards/DCM/Src/App/states/App_DriveState.c

```diff
@@ -1,20 +1,62 @@
+#include <stdlib.h>
+#include <math.h>
 #include "states/App_AllStates.h"
 #include "states/App_InitState.h"
 #include "App_SetPeriodicCanSignals.h"
 
-// TODO: Have separate maximum torque requests for motoring and generating
 #define MAX_TORQUE_REQUEST_NM (90.0f)
 
-void App_SetPeriodicCanSignals_TorqueRequests(struct DcmCanTxInterface *can_tx, struct DcmCanRxInterface *can_rx)
+#define MIN(a, b) ((a) < (b) ? (a) : (b))
+#define RPM_TO_RADS(rpm) ((rpm) * (float)M_PI / 30.0f)
+
+struct EfficiencyLUTEntry
 {
-    float torque_request = 0.0f;
+    float speed_rpm;
+    float efficiency_estimate;
+};
+
+struct EfficiencyLUTEntry EfficiencyLUT[14] = {
```

Comment:
the .speed_rpm part of the struct seems redundant here. You could do the same thing as a simple 1-D array instead of a struct.

uint8_t eff_lut_index = speed_rpm/RPM_INTERVAL; //RPM_INTERVAL is 500 in the LUT you proposed
float efficiency_estimate = EfficiencyLUT[eff_lut_index ];

This way you're not iterating through a for loop to find the right index, and you don't need to waste memory on the struct. Will become important when you're dealing with a giant 2-D LUT with interpolation

Reply:
in the case that we have time to test the efficiency a bit better we can quickly update their values in the LUT so i dont mind keeping this for now in a header file. In our implementation in the state machine we can do what you suggested